liblumen_alloc/src/erts/term/map.rs

```diff
@@ -55,6 +55,12 @@ impl Map {
         self.value.len()
     }
 
+    pub fn remove(&self, key: Term) -> HashMap<Term, Term> {
+        let mut map = self.value.clone();
```

Comment:
This could be further optimized by only cloning if the key is actually being removed.  If you change the return type to a `bool`, you could also signal to the caller whether removal occurred and if it is false, then you can return the same `Term` you were passed in, which would save you from allocating a new, identical, map on process heap.

Reply:
Couldn't change the type to bool, because if it could be removed, I needed to return the new map without that value. So, I went with an `Option<HashMap>`, where `None` means nothing was removed and then it returns the existing map.